apps/sparo-lib/src/cli/commands/checkout.ts

```diff
@@ -74,7 +76,24 @@ export class CheckoutCommand implements ICommand<ICheckoutCommandOptions> {
   ): Promise<void> => {
     const { _gitService: gitService } = this;
     terminalService.terminal.writeDebugLine(`got args in checkout command: ${JSON.stringify(args)}`);
-    const { b, B, branch, startPoint } = args;
+    const { b, B, startPoint } = args;
+
+    let branch: string | undefined = args.branch;
+
+    /**
+     * Special case: sparo checkout -
+     * yargs can not handle this, patch this case here.
+     */
+    if (!branch) {
+      const checkoutIndex: number = process.argv.findIndex((value: string) => value === 'checkout');
+      if (checkoutIndex >= 0 && process.argv[checkoutIndex + 1] === '-') {
+        branch = '-';
+        // FIXME: supports "sparo checkout -"
```

Comment:
I investigated this problem. Yargs does have a configuration option `.parserConfiguration({'unknown-options-as-args': true})` that in theory would allow `-`, however the parser is hardcoded to ignore `match(/^-/)` and convert `-` to `defaultValue(key)` which is an empty string:

[yargs-parser.ts](https://github.com/yargs/yargs-parser/blob/3aba24ceaa1a06ceb982c63a06002526d781e826/lib/yargs-parser.ts#L286)
```ts
    else {
        next = args[i + 1];
        if (next !== undefined && (!next.match(/^-/) ||
            next.match(negative)) &&
            !checkAllAliases(key, flags.bools) &&
            !checkAllAliases(key, flags.counts)) {
            setArg(key, next);
            i++;
        }
        else if (/^(true|false)$/.test(next)) {
            setArg(key, next);
            i++;
        }
        else {
            setArg(key, defaultValue(key));
        }
    }
```

In any case, the ideal Yargs feature wouldn't be "unknown options" but rather "special strings" that should always be interpreted literally, since Git's `-` really is special compared to all other CLI punctuation.

A safe workaround can be like this:

```ts
const normalizedArgs = [...process.argv];

// If defined, this is a unique token that cannot conflict with any real argument
let dashToken = undefined

// Does argv include "-"
for (let i = 2; i<normalizedArgs.length; ++i) {
  if (normalizedArgs[i] === '-') {
    if (dashToken === undefined) {
      // Generate a dashToken guaranteed to be free of conflicts
      const existingArgs = new Set(normalizedArgs);
      for (let j = 1; ; ++j) {
        dashToken = `DASH_${j}`;
        if (!existingArgs.has(dashToken)) {
          break;
        }
      }
    }
    // replace "-" with dashToken
    normalizedArgs[i] = dashToken;
  }
}
```

Then you would search+replace dashToken in certain contexts:
- When printing CLI parameters on the console
- When generating the `argv` to invoke the `git` subprocess.

This solution is simple enough that you could probably include it in this PR.  It will also solve other subcommands such as `git apply` and `git commit`.